### PR TITLE
More Modcomp shit

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Adds in a basic upload server system, which I will use in the future for an overhaul of the Archivist area. Also deletes some useless programs (e.g. newscaster) and assets (e.g. old PDA images). General code cleanup in the modcomp system.

## Why It's Good For The Game

In the short term, I can see 2 main uses for this basic database system.
1. Moving all of the physical paper documents on SCPs to a server room, saving on resources as well as allowing for more interesting gameplay.
2. Moving downloadable programs from a transient online service to a server room, allowing for sabotage and better control over file access.

## Future Improvements

Obviously, the upload database should be given more use in the future.
I didn't make any sprites for the two new programs.

## Changelog

:cl:
add: Adds new database server and client programs, used to host a basic file browser service.
del: Removed newscaster program and flight program reports
imagedel: Deleted newspaper images and old PDA images.
spellcheck: Changed some baystation references to SCP references.
code: General code cleanup in modcomp files as well as some helper procs.
/:cl:

### CODE REVIEW

**Comment by TheDarkElites** [comment]
Omg no way hes gonna finish mod comps stuff!!!

We stan cheese!!!

File: code/modules/modular_computers/file_system/programs/servers/upload_database.dm [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	usage_flags = PROGRAM_CONSOLE
+	tgui_id = "SCiPUploadServer"
+
+	var/list/files_required_access = list()									// associative list - the key is the filename, and the value is a list of required accesses
+	var/list/enabled_files = list()											// list of every file that's available for download. filenames, not the actual file datum
+	var/editing_file														// which file we're currently changing access requirements for
+	var/hosting	= FALSE														// if we're available on SCiPnet for use
+	var/server_name = ""													// user-set name for easier identification by the client
+	var/unique_token 														// UID of this server
+	var/list/datum/computer_file/program/upload_database_c/clients = list()	// all connected clients
```
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database.dm]: Use the auto doc feature for these.
```
/// comment
var/whatever_the_fuck
```

File: code/modules/modular_computers/file_system/programs/servers/upload_database.dm [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+			if (get_access_desc(access))
+				accesses.Add(list(list(
+					"desc" = replacetext(get_access_desc(access), " ", "&nbsp"),
+					"ref" = access,
+					"allowed" = (access in id_card.access) ? 1 : 0)))
+
+		regions.Add(list(list(
+			"name" = get_region_accesses_name(i),
+			"accesses" = accesses)))
+	data["regions"] = regions*/
```
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database.dm]: ???

File: code/modules/modular_computers/file_system/programs/servers/upload_database.dm [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+	return data
+
+/datum/computer_file/program/upload_database/tgui_act(action, list/params, datum/tgui/ui)
+	if(..())
+		return TRUE
+
+	//var/obj/item/stock_parts/computer/hard_drive/HDD = computer.hard_drive
```
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database.dm]: more stuff you left commented out

File: code/modules/modular_computers/file_system/programs/servers/upload_database_c.dm [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	network_destination = "other device via server-to-client tunnel"
+	available_on_ntnet = TRUE
+	tgui_id = "SCiPUploadClient"
+
+	var/error = ""											// Error screen
+	var/datum/computer_file/program/upload_database/remote	// which server are we currently downloading from?
+	var/list/datum/computer_file/available_files = list()	// all the files on the remote (that we can see)
+	var/datum/computer_file/downloading_file				// the file we're currently downloading
+	var/download_completion = 0								// download progress in GQ
+	var/actual_netspeed = 0									// actual transfer speed (displayed in UI)
```
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database_c.dm]: Use auto doc here too

File: code/modules/modular_computers/file_system/programs/servers/upload_database_c.dm [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+		actual_netspeed = min((remote.ntnet_speed / (remote.clients?.len ? remote.clients?.len : 1)), ntnet_speed)
+		download_completion += actual_netspeed
+
+		if(download_completion >= downloading_file.size)
+			if(!computer || !computer.hard_drive || !computer.hard_drive.store_file(downloading_file))	// see if we can download the file
+				crash("I/O Error: Unable to save file. Check your hard drive and try again.")
+			else
+				cleanup_download()
+
+// crashes the client and displays specific error message
```
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database_c.dm]: Auto doc for the proc defines aswell

**Review by TheDarkElites — CHANGES_REQUESTED** [review]
Stuff you should fix
Also, you should get Tusurpta to look at this after you fix the things I mentioned since he's a much better maintainer than me.

**Comment by TheDarkElites** [comment]
This looks good enough for me, will assign it as waiting for test merge.

**Comment by TheDarkElites** [comment]
[17:06:33] Runtime in virus.dm, line 154: Cannot read null.access
proc name: process tick (/datum/computer_file/program/virus/process_tick)
src: /datum/computer_file/program/v... (/datum/computer_file/program/virus)
call stack:
/datum/computer_file/program/v... (/datum/computer_file/program/virus): process tick()
the laptop computer (/obj/item/modular_computer/laptop/preset/custom_loadout/advanced): Process(10, 4953, Objs (/datum/controller/subsystem/processing/obj))
Objs (/datum/controller/subsystem/processing/obj): fire(0)
Objs (/datum/controller/subsystem/processing/obj): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)

File: code/game/movietitles.dm [hunk 6]
```diff
@@ -124,8 +124,8 @@ GLOBAL_LIST(end_titles)
 		if(!cast.len && !chunksize)
 			chunk += "CAST:"
 		var/job = ""
-		if(GetAssignment(H) != "Unassigned")
-			job = ", [uppertext(GetAssignment(H))]"
+		if(H.get_assignment("NONE", "NONE") != "NONE")
```
Comment (TheDarkElites) [inline code/game/movietitles.dm:127]: get rid of the global GetAssignment proc all together if you're gonna replace all of the use cases.

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

File: code/game/movietitles.dm (hunk 6 above)
Comment (TheDarkElites) [reply code/game/movietitles.dm:127]: nevermind im fucking blind apparently

File: code/modules/modular_computers/file_system/reports/crew_record.dm [hunk 7]
```diff
@@ -132,12 +131,26 @@ GLOBAL_VAR_INIT(arrest_security_status, "Arrest")
...
+			if(F.needs_big_box)
+				dat += "<tr>"
+			dat += "<td>[F.get_value()]"
+	dat += "</tt>"
+	return dat
+
 // Global methods
 // Used by character creation to create a record for new arrivals.
-/proc/CreateModularRecord(mob/living/carbon/human/H)
+/mob/proc/CreateModularRecord()
```
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/reports/crew_record.dm:149]: move above //Global methods comment since this is now a local proc

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

File: code/modules/modular_computers/file_system/reports/crew_record.dm (hunk 7 above)
Comment (aidan-bowman) [reply code/modules/modular_computers/file_system/reports/crew_record.dm:149]: done (nerd)

**Review by TheDarkElites — APPROVED** [review]
Tsurpta said this is good, it looks good to me, and it passed TM

god bless the modcomp pr

**Pull request merged**